pkg/apis/k0s/v1beta1/api.go

```diff
@@ -100,10 +104,20 @@ func (a *APISpec) getExternalURIForPort(port int) string {
 	return fmt.Sprintf("https://%s:%d", addr, port)
 }
 
+// APIServerAddress returns the address the API is listening on
+func (a *APISpec) APIServerAddress() string {
+	return a.getAPIServerAddress(a.BindAddress)
+}
+
+func (a *APISpec) getAPIServerAddress(address string) string {
+	return address
+}
+
```

Comment:
OTOH, if this would fall back to `localhost` if BindAddess is empty, it would make some sense again, and it would also mean that there'd be no behavioral change at all when BindAddress is not specified.